### PR TITLE
Fix "MoveStaticMembers" for VB with invalid member

### LINKED ISSUE #66489: INvalidOperationException from VisualBasicMoveStaticMembersRefactoringProvider
image

```
System.InvalidOperationException : Operation is not valid due to the current state of the object.
   at Microsoft.CodeAnalysis.Shared.Extensions.SemanticModelExtensions.GetRequiredDeclaredSymbol(SemanticModel semanticModel,SyntaxNode declaration,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MoveStaticMembers.AbstractMoveStaticMembersRefactoringProvider.<>c__DisplayClass1_0.<ComputeRefactoringsAsync>b__0(SyntaxNode m)
   at System.Collections.Immutable.ImmutableArray.CreateRange[TSource,TResult](ImmutableArray`1 items,Func`2 selector)
   at async Microsoft.CodeAnalysis.MoveStaticMembers.AbstractMoveStaticMembersRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
```

After pasting C# code into VB file in Roslyn

```
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.


Namespace Microsoft.CodeAnalysis.CSharp;

/// <summary>
/// Manages instrumentation state.
/// </summary>
internal sealed Class InstrumentationState
{
    /// <summary>
    /// Used to temporarily suspend instrumentation, for example when lowering expression tree.
    /// </summary>
    Public bool IsSuppressed { Get; Set; }

    /// <summary>
    /// Current instrumenter.
    /// </summary>
    Public Instrumenter Instrumenter { Get; Set; } = Instrumenter.NoOp;

    Public void RemoveCodeCoverageInstrumenter()
    {
        Instrumenter = recurse(Instrumenter);

        Static Instrumenter recurse(Instrumenter instrumenter)
            => instrumenter switch
            {
                CodeCoverageInstrumenter { Previous: var previous } => recurse(previous),
                CompoundInstrumenter compound => compound.WithPrevious(recurse(compound.Previous)),
                _ => Instrumenter,
            };
    }
}
```

### PR DESCRIPTION
**Pull request by ryzngard**
Don't error if a member isn't found relating to code. 

Fixes #66489

### CODE REVIEW

File: src/Features/Core/Portable/MoveStaticMembers/AbstractMoveStaticMembersRefactoringProvider.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var memberNodeSymbolPairs = selectedMemberNodes
-                .SelectAsArray(m => (node: m, symbol: semanticModel.GetRequiredDeclaredSymbol(m, cancellationToken)))
+                .SelectAsArray(m => (node: m, symbol: semanticModel.GetDeclaredSymbol(m, cancellationToken)))
                 // Use same logic as pull members up for determining if a selected member
                 // is valid to be moved into a base
                 .WhereAsArray(pair => MemberAndDestinationValidator.IsMemberValid(pair.symbol) && pair.symbol.IsStatic);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/MoveStaticMembers/AbstractMoveStaticMembersRefactoringProvider.cs:43]: does IsMemberValid ensure that symbol is nonnull?

File: src/Features/Core/Portable/PullMemberUp/MemberAndDestinationValidator.cs [hunk 2]
```diff
@@ -27,8 +26,13 @@ public static bool IsDestinationValid(Solution solution, INamedTypeSymbol destin
             return isDestinationInSourceAndNotGeneratedCode;
         }
 
-        public static bool IsMemberValid(ISymbol member)
+        public static bool IsMemberValid([NotNullWhen(true)] ISymbol? member)
         {
+            if (member is null)
+            {
+                return false;
+            }
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/PullMemberUp/MemberAndDestinationValidator.cs:34]: yes.  yes it does.

File: src/Features/VisualBasic/Portable/CodeRefactorings/MoveStaticMembers/VisualBasicMoveStaticMembersRefactoringProvider.vb [hunk 3]
```diff
@@ -13,9 +13,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.MoveStaticMembers
     Friend Class VisualBasicMoveStaticMembersRefactoringProvider
         Inherits AbstractMoveStaticMembersRefactoringProvider
 
+#Disable Warning RS0033 ' Importing constructor should be marked with 'ObsoleteAttribute'
         <ImportingConstructor>
-        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
         Public Sub New()
+#Enable Warning RS0033 ' Importing constructor should be marked with 'ObsoleteAttribute'
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasic/Portable/CodeRefactorings/MoveStaticMembers/VisualBasicMoveStaticMembersRefactoringProvider.vb]: but why?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/VisualBasic/Portable/CodeRefactorings/MoveStaticMembers/VisualBasicMoveStaticMembersRefactoringProvider.vb (hunk 3 above)
Comment (ryzngard) [reply src/Features/VisualBasic/Portable/CodeRefactorings/MoveStaticMembers/VisualBasicMoveStaticMembersRefactoringProvider.vb]: removed. I was annoyed with tests until I found a better way

File: src/Features/Core/Portable/PullMemberUp/MemberAndDestinationValidator.cs (hunk 2 above)
Comment (ryzngard) [reply src/Features/Core/Portable/PullMemberUp/MemberAndDestinationValidator.cs:34]: NRT ftw :)

**Pull request merged**